### ISSUE TITLE
Incorrect resolution during locals phase

### ISSUE THREAD

**Issue by dominicbarnes** [componentjs/resolver.js#12]
I'm not sure if this problem technically falls under resolver or remotes, so we can move this if needed.

I'm getting a "invalid semver range" error when I should not be. After the resolver transitions from resolving remotes/dependencies to resolving locals, it only knows about versions discovered during the remote phase.

For example, a lot of components depend on component/classes, but the ones I'm happening to use all pin version `1.1.*`, when I need a feature I introduced in `1.2.0`. When the resolver gets to my local dependencies, where I'm trying to get `^1.2.0`, I get:

```
   error : invalid semver range "^1.2.0" for "component/classes"
```

Basically, if a "local remote" fails to find a matching version locally, it should test against remote again to see if any matching versions are available.
